lib/internal/readline/interface.js

```diff
@@ -1024,9 +1095,19 @@ class Interface extends InterfaceConstructor {
    *   }}
    */
   getCursorPos() {
-    const strBeforeCursor =
-      this[kPrompt] + StringPrototypeSlice(this.line, 0, this.cursor);
-    return this[kGetDisplayPos](strBeforeCursor);
+    // Handle multiline input by accounting for "| " prefixes on continuation lines
+    const lineUptoCursor = StringPrototypeSlice(this.line, 0, this.cursor);
+    const lines = StringPrototypeSplit(lineUptoCursor, '\n');
+
+    // First line uses normal prompt, subsequent lines use "| " (2 chars)
+    let displayText = this[kPrompt] + lines[0];
+
+    // Add continuation lines with their prefixes
+    for (let i = 1; i < lines.length; i++) {
+      displayText += '\n| ' + lines[i];
+    }
+
+    return this[kGetDisplayPos](displayText);
```

Comment:
I don't think this should be changed. Instead, if `_getDisplayPos()` would handle this properly, it would just work in all cases. I believe I commented in another file almost the same.